aii-opennebula/src/main/perl/opennebula.pm

```diff
@@ -62,18 +62,22 @@ sub make_one
         $rpc = $domainname;
         $main::this_app->info ("Detected configfile RPC section: [$rpc]");
     };
-    my $port = $config->{$rpc}->{port} || 2633;
-    my $host = $config->{$rpc}->{host} || "localhost";
+    my $port = $config->{$rpc}->{port};
+    my $host = $config->{$rpc}->{host};
+    my $url = $config->{$rpc}->{url} || "http://localhost:2633";
     my $user = $config->{$rpc}->{user} || "oneadmin";
     my $password = $config->{$rpc}->{password};
 
+    $url = "http://$host:2633" if ($host);
```

Comment:
this overwrites the url set by `$config->{$rpc}->{url}` if only $host is set?
One should probably specify `url NAND (host OR port)`  since you have defaults, otherwise this can be ambiguous, (and I would throw an error in that case..) Then, if the `url` is set, take that one, if not keep old behavior to keep backwards compatibility ?